artemis/src/main/java/com/artemis/InvocationStrategy.java

```diff
@@ -7,11 +7,10 @@
 	protected void process(Bag<BaseSystem> systems) {
 		Object[] systemsData = systems.getData();
 		for (int i = 0, s = systems.size(); s > i; i++) {
-			updateEntityStates();
-
 			BaseSystem system = (BaseSystem) systemsData[i];
 			if (!system.isPassive()) {
 				system.process();
+				updateEntityStates();
```

Comment:
hmm... but it still requires calling updateEntityStates separately, if the system create/edits/deletes entities. Tempting to deprecate passive() - it should be handled by the invocation strategy instead. Would mess with a lot of code bases though.